fireapp/lib/presentation/calendar/calendar_widget.dart

```diff
@@ -44,20 +35,346 @@ class _CalendarState extends FireAppState<CalendarView>
+                  textAlign: TextAlign.center,
+                  style: const TextStyle(
+                      fontSize: 20, fontWeight: FontWeight.bold),
+                ),
+              ],
+            ),
+          ),
+          Expanded(
+            child: Column(children: dayEvents),
+          ),
+        ],
+      ));
+    }
 
+    return Column(children: eventWidgets);
+  }
+
+  Widget _buildEventCard(CalendarEvent event, int index, int totalEvents) {
+    // Parse parameters from event
+    final unavailabilityEvent = event.event;
+    final title = unavailabilityEvent.title;
+    final displayTime = event.displayTime;
+    final eventId = unavailabilityEvent.eventId;
+    final cardColor = getColorForEvent(eventId.toString());
+
+    return Card(
+      color: cardColor,
+      margin: const EdgeInsets.symmetric(vertical: 4, horizontal: 8),
+      child: ListTile(
+        title: Text(title),
+        subtitle: Text(displayTime),
+        trailing: PopupMenuButton<String>(
+          onSelected: (value) {},
+          itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
+            PopupMenuItem<String>(
+                value: 'Edit',
+                onTap: () {
+                  viewModel.editEventNavigate(unavailabilityEvent);
+                },
+                child: Text(
+                    AppLocalizations.of(context)?.calendarItemEdit ?? "Edit")),
+            PopupMenuItem<String>(
+              value: 'Delete',
+              child: Text(
+                  AppLocalizations.of(context)?.calendarItemDelete ?? "Delete"),
+              onTap: () {
+                viewModel.deleteUnavailability(eventId);
+              },
+            ),
+          ],
+        ),
+      ),
+    );
   }
 
   @override
   Widget build(BuildContext context) {
-    return ScrollViewBottomContent(
-      padding: EdgeInsets.all(1.rdp()),
-      bottomChildren: const [
-        FillWidth(child: Placeholder()),
-      ],
-      children: const [Text("Calendar placeholder")],
+    return Scaffold(
+      appBar: AppBar(
+        title: TextButton(
+          onPressed: () => _showMonthPicker(context),
+          child: Text(
+            _selectedMonthLabel,
+            style: const TextStyle(
+                color: Colors.black, fontWeight: FontWeight.bold, fontSize: 24),
+          ),
+        ),
+        backgroundColor: Colors.white,
+      ),
+      body: SafeArea(
+        child: StreamBuilder<List<CalendarEvent>>(
+          stream: viewModel.displayEventsStream,
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.waiting) {
+              return const Center(child: CircularProgressIndicator());
+            }
+            if (!snapshot.hasData || snapshot.data!.isEmpty) {
+              return Center(
+                  child: Text(
+                      AppLocalizations.of(context)?.calendarNoItemsLabel ??
+                          "No events to display"));
```

Comment:
Brackets should be on newlines to better visualise hierarchy.

Reply:
another dart format issue, will make this change anyway